File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> num_index;

        for (size_t i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
            if (num_index.count(complement)) {
                return {num_index[complement], (int)i};
            }
            num_index[nums[i]] = i;
        }

        return {};
    }
};

File: 0001-two-sum/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen = HashMap::new();

        for (index, num) in nums.iter().enumerate() {
            let other = target - num;

            if seen.contains_key(&other) {
                return vec![seen[&other], index as i32];
            }

            seen.insert(num, index as i32);
        }

        return Vec::new();
    }
}
